analysis/test/association_test.R

```diff
@@ -199,7 +199,43 @@ TestComputeDistributionEM <- function() {
                                 params, marginals = NULL,
                                 estimate_var = FALSE)
 
-  checkTrue(abs(dist$fit[1, 1] - 0.5) < 0.15)
+  checkTrue(abs(dist$fit["1", "1"] - 0.5) < 0.15)
+  checkTrue(abs(dist$fit["2", "2"] - 0.5) < 0.15)
+
+  # Test 6: Check the computed joint distribution with randomized
+  # correlated inputs from the Poisson distribution
+  # Expect to have correlation between strings n and n + 1
+  N <- 1000
+  params <- list(k = 16, h = 2, m = 4, p = 0.1, q = 0.9, f = 0.1)
+  variable_opts <- list(deterministic = FALSE, independent = FALSE)
+  sim <- Simulate(N, num_variables = 2, params, variable_opts)
+  dist <- ComputeDistributionEM(sim$reports, sim$cohorts,
+                                sim$maps, ignore_other = TRUE,
+                                params, marginals = NULL,
+                                estimate_var = FALSE)
+
+  print_dist <- TRUE  # to print joint distribution, set to TRUE
+
+  if (print_dist) {
+    # dist$fit[dist$fit<1e-4] <- 0
+    # Sort by row names and column names to visually see correlation
+    print(dist$fit[sort(rownames(dist$fit)), sort(colnames(dist$fit))])
+  }
+  
+  # Check for correlations (constants chosen heuristically to get good
+  # test confidence with small # of samples)
+  # Should have mass roughly 1/2e and 1/2e each
+  checkTrue(abs(dist$fit["1", "1"] - dist$fit["1", "2"]) < 0.05)
+  checkTrue(abs(dist$fit["2", "2"] - dist$fit["2", "3"]) < 0.05)
+  
+  # Should have mass roughly 1/4e and 1/4e each
+  checkTrue(abs(dist$fit["3", "3"] - dist$fit["3", "4"]) < 0.03)
+
+  # Check for lack of correlation
+  checkTrue(dist$fit["1", "3"] < 0.01)
+  checkTrue(dist$fit["2", "1"] < 0.01)
+  checkTrue(dist$fit["3", "1"] < 0.01)
+  checkTrue(dist$fit["3", "2"] < 0.01)
 
```

Comment:
For completeness sake, does it make sense to add checks that dist$fit["1", "4"] and dist$fit["2", "4"] are close to zero too?